Generated code does not compile with latest nightly-build

### Description

windows nightly build, see: https://github.com/OpenModelica/OpenModelicaLibraryTesting/actions/runs/11067454279/job/30751855594

`HelloWorld.c:163:5: error: incompatible function pointer types initializing 'const void (*)(DATA *, threadData_t *)' (aka 'const void (*)(struct DATA *, struct threadData_s *)') with an expression of type 'void (DATA *, threadData_t *)' (aka 'void (struct DATA *, struct threadData_s *)') [-Wincompatible-function-pointer-types]`

--- a/OMCompiler/Compiler/Template/CodegenC.tpl
+++ b/OMCompiler/Compiler/Template/CodegenC.tpl
@@ -4418,7 +4418,7 @@ template functionXXX_system(list<SimEqSystem> eqs, String name, Integer n, Strin
   {
     int id;
 
-    static const void (*eqFunctions[<%nFuncs%>])(DATA*, threadData_t*) = {
+    static void (*const eqFunctions[<%nFuncs%>])(DATA*, threadData_t*) = {
       <%eqs |> eq => '<%symbolName(modelNamePrefix, "eqFunction")%>_<%equationIndexGeneral(eq)%>'; separator=",\n"%>
     };
 

